test/integration/test-transaction.js

```diff
@@ -0,0 +1,69 @@
+"use strict";
+
+const base = require("../base.js");
+const assert = require("chai").assert;
+
+describe("transaction", function() {
+
+
+  before(function(done) {
+    shareConn.query(
+      "CREATE TEMPORARY TABLE testTransaction (v varchar(10))",
+      err => {
+        if (err) return done(err);
+        done();
+      }
+    );
+  });
+
+  it("transaction rollback", function(done) {
+    shareConn.rollback();
+    shareConn.query("SET autocommit=0", () => {
+      assert.equal(shareConn.info.status, 0);
+      shareConn.beginTransaction((err) => {
+        if (err) done(err);
+        assert.equal(shareConn.info.status, 1);
+        shareConn.query("INSERT INTO testTransaction values ('test')", (err) => {
+          if (err) done(err);
+          assert.equal(shareConn.info.status, 1);
+          shareConn.rollback((err) => {
+            if (err) done(err);
+            assert.equal(shareConn.info.status, 0);
+            shareConn.query("SELECT count(*) as nb FROM testTransaction", (err, rows) => {
+              if (err) done(err);
+              assert.equal(shareConn.info.status, 33);
```

Comment:
Would be better to have constants at some point, instead of 1, 33 other magical numbers